packages/contracts-bedrock/test/L1/opcm/OPContractsManagerV2.t.sol

```diff
@@ -9,20 +9,200 @@ import { DisputeGames } from "test/setup/DisputeGames.sol";
 import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
 import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
 import { ISystemConfig } from "interfaces/L1/ISystemConfig.sol";
 import { IOPContractsManagerStandardValidator } from "interfaces/L1/IOPContractsManagerStandardValidator.sol";
 import { IOPContractsManagerV2 } from "interfaces/L1/opcm/IOPContractsManagerV2.sol";
 
+/// @title OPContractsManagerV2_TestInit
+/// @notice Base test initialization contract for OPContractsManagerV2.
+contract OPContractsManagerV2_TestInit is CommonTest, DisputeGames {
+    /// @notice Fake prestate for Cannon games.
+    Claim cannonPrestate = Claim.wrap(bytes32(keccak256("cannonPrestate")));
+
+    /// @notice Fake prestate for Cannon Kona games.
+    Claim cannonKonaPrestate = Claim.wrap(bytes32(keccak256("cannonKonaPrestate")));
+
+    /// @notice Special string constant used to indicate that we expect a revert without any data.
+    bytes public constant EXPECT_REVERT_WITHOUT_DATA = bytes("EXPECT_REVERT_WITHOUT_DATA");
+
+    /// @notice Buffer percentage (relative to EIP-7825 gas limit) allowed for deployments.
+    uint256 public constant DEPLOY_GAS_BUFFER_PERCENTAGE = 80; // 80%
+
+    /// @notice Sets up the test suite.
+    function setUp() public virtual override {
+        super.setUp();
+        skipIfDevFeatureDisabled(DevFeatures.OPCM_V2);
+    }
+
+    /// @notice Helper function that runs an OPCM V2 deploy, asserts that the deploy was successful,
+    ///         and runs post-deploy standard validator checks.
+    /// @param _opcm The OPCM contract to use for deployment.
+    /// @param _deployConfig The full config for the deployment.
+    /// @param _revertBytes The bytes of the revert to expect (empty if no revert expected).
+    /// @param _expectedValidatorErrors The StandardValidator errors to expect.
+    /// @return cts_ The deployed chain contracts.
+    function _runOpcmV2DeployAndChecks(
+        IOPContractsManagerV2 _opcm,
+        IOPContractsManagerV2.FullConfig memory _deployConfig,
+        bytes memory _revertBytes,
+        string memory _expectedValidatorErrors
+    )
+        internal
+        returns (IOPContractsManagerV2.ChainContracts memory cts_)
+    {
+        // Grab the proposer and challenger from deploy config for validator.
+        address deployProposer;
+        address deployChallenger;
+        for (uint256 i = 0; i < _deployConfig.disputeGameConfigs.length; i++) {
+            if (_deployConfig.disputeGameConfigs[i].gameType.raw() == GameTypes.PERMISSIONED_CANNON.raw()) {
+                IOPContractsManagerV2.PermissionedDisputeGameConfig memory parsedArgs = abi.decode(
+                    _deployConfig.disputeGameConfigs[i].gameArgs, (IOPContractsManagerV2.PermissionedDisputeGameConfig)
+                );
+                deployProposer = parsedArgs.proposer;
+                deployChallenger = parsedArgs.challenger;
+                break;
```

Comment:
You shouldn't need the loop and break since the order [is enforced](https://github.com/ethereum-optimism/optimism/blob/8852811f38bbd321406d3cdfad07553c42571661/packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol#L619), but this is more robust I suppose.